EncoreTickets.SDK/Api/Results/Exceptions/ApiException.cs

```diff
@@ -93,77 +88,75 @@ public ApiException(string message)
             Response = response;
         }
 
-        protected string GetMessage()
+        /// <summary>
+        /// Returns easily read errors that are the cause of the exception.
+        /// </summary>
+        /// <returns>Collection of strings</returns>
+        protected List<string> GetErrors()
         {
-            if (string.IsNullOrEmpty(predefinedMessage))
-            {
-                return Errors.Any() ? string.Join("; ", Errors) : DefaultMessage;
-            }
-
-            return predefinedMessage;
+            var errors = GetErrorsAsString(Response, ContextInResponse);
+            return FilterErrorsAsStrings(errors);
         }
 
         /// <summary>
-        /// Returns easily read errors that are the cause of the exception.
+        /// Returns a string contained easily read info about errors that are the cause of the exception.
         /// </summary>
-        /// <returns></returns>
-        protected List<string> GetErrors()
+        /// <returns>Message about errors</returns>
+        protected string GetMessage()
         {
-            if (ContextInResponse?.Errors == null)
+            if (predefinedMessage != null)
             {
-                return new List<string>
-                {
-                    string.IsNullOrEmpty(Response.StatusDescription)
-                        ? Response.ErrorMessage
-                        : Response.StatusDescription
-                };
+                return predefinedMessage;
             }
 
-            var contextErrors = ContextInResponse.Errors.Select(ConvertErrorToString)
-                .Where(x => !string.IsNullOrEmpty(x));
-            return contextErrors.ToList();
+            return Errors != null && Errors.Any()
+                ? string.Join("; ", Errors)
+                : DefaultMessage;
         }
 
-        private string ConvertErrorToString(Error error)
+        protected List<string> FilterErrorsAsStrings(IEnumerable<string> errors)
         {
-            var message = error.Message;
-            if (!string.IsNullOrEmpty(error.Field))
+            if (errors == null)
             {
-                message = $"{error.Field} - {message}";
+                return null;
             }
 
-            return message;
+            var notEmptyErrors = errors.Where(x => !string.IsNullOrEmpty(x)).ToList();
+            return notEmptyErrors.Any() ? notEmptyErrors : null;
```

Comment:
Are you sure it's a good idea to default to null if there are no elements satisfying the condition?

Reply:
I think it is better to always have the same result if there are no elements. Do you think it's better to use an empty enumerable than a null? I do not see a problem in returning null

Reply:
My main problem here is that as per my another comment, this method looks like it could be moved to IEnumerable extensions. Currently it looks like a wrapper around a very common filter ```.Where(x => !string.IsNullOrEmpty(x))```. If we move it to other extensions and only leave the Where, we could start using it like any other Linq method. We could also chain it with other calls and preserve the deferred execution. For example,

```
source.Where(x => x?.Foo >= 0)
    .OrderBy(x => x => x.Bar)
    .Select(x => x.StringValue)
    .FilterEmptyStrings()
    .FirstOrDefault();
```
Here we would only need to worry about NullReferenceException on the first call to Where.

Even without it, I think it would be better if we could differentiate between "The collection is null" and "Everything in the collection is null or empty" unless you have a reason why we don't need this